functions/src/emailNotifications/utils.ts

```
@@ -42,35 +52,57 @@ export const getUserAndEmail = async (userName: string) => {
   return { toUser, toUserEmail }
 }
 
-export const isValidEmailCreationRequest = async ({
-  email,
-  toUserName,
-}: IMessageDB) => {
-  const user = await firebaseAuth.getUserByEmail(email)
-  if (user.email !== email) {
-    throw new Error(
-      `Emailing of new message blocked: Email address not attached to user record`,
-    )
-  }
-
+export const isBelowMessageLimit = async (email) => {
   const { docs } = await db
     .collection(DB_ENDPOINTS.messages)
     .where('email', '==', email)
     .get()
   if (docs.length >= EMAIL_ADDRESS_SEND_LIMIT) {
-    throw new Error(
-      `Emailing of new message blocked: User exceeded email message count`,
-    )
+    throw new Error(errors.MESSAGE_LIMIT)
   }
+  return true
+}
 
-  const { toUser } = await getUserAndEmail(toUserName)
+export const isReceiverContactable = async (userName) => {
+  const { toUser } = await getUserAndEmail(userName)
   if (!toUser.isContactableByPublic) {
-    throw new Error(`Emailing of new message blocked: Profile not contactable`)
+    throw new Error(errors.PROFILE_NOT_CONTACTABLE)
+  }
+  return true
+}
+
+export const isSameEmail = (userDoc, email) => {
+  if (userDoc.email !== email) {
+    throw new Error(errors.NO_ATTACHED_USER)
   }
+  return true
+}
 
+export const isUserAllowedToMessage = async (uid) => {
+  const { toUser } = await getUserAndEmail(uid)
+  if (toUser.isBlockedFromMessaging) {
+    throw new Error(errors.USER_BLOCKED)
+  }
   return true
 }
 
+export const isValidEmailCreationRequest = async ({
+  email,
+  toUserName,
+}: IMessageDB) => {
+  const userDoc = await firebaseAuth.getUserByEmail(email)
+
+  try {
+    isSameEmail(userDoc, email)
+    await isUserAllowedToMessage(userDoc.uid)
+    await isBelowMessageLimit(email)
+    await isReceiverContactable(toUserName)
+    return true
+  } catch {
+    return Promise.reject(false)
```

we never expose the error messages attached to each fn! two proposed alternatives: 
1. catch the error here and log it
2. (preferred) let the error be thrown and it will be caught and logged through the `withErrorAlerting` hook